src/query/block/column.go

```diff
@@ -223,10 +221,54 @@ func (cb ColumnBlockBuilder) AddCols(num int) error {
 	return nil
 }
 
+// PopulateColumns sets all columns to the given row size.
+func (cb ColumnBlockBuilder) PopulateColumns(size int) {
+	for i := range cb.block.columns {
+		cb.block.columns[i] = column{Values: make([]float64, size)}
+	}
+}
+
+// SetRow sets a given block row to the given values and metadata.
+func (cb ColumnBlockBuilder) SetRow(
+	idx int,
+	values []float64,
+	meta SeriesMeta,
+) error {
+	cols := cb.block.columns
+	if len(values) == 0 {
+		// Sanity check. Should never happen.
+		return errors.New("cannot insert empty values")
+	}
+
+	if len(values) != len(cols) {
+		return fmt.Errorf("inserting column size %d does not match column size: %d",
+			len(values), len(cols))
+	}
+
+	rows := len(cols[0].Values)
```

Comment:
Yeah this thing is a bit awkward (been meaning to update it), but basically it has a row per series; since the series are expected to be in steps by this point, it's safe to say they're the same size here. 

Been meaning to refactor this file (and a bunch of other workflows)